### PR TITLE
Refactor NumPy-compatible elemwise broadcast operators

### PR DESCRIPTION
**Pull request by haojin2**
## Description ##
Adding `bitwise_xor` on the side.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Refactor and split files
- [x] `bitwise_xor` operator
- [x] Unit tests

## Comments ##
`bitwise_xor` kernel was contributed by @gyshi.

### CODE REVIEW

File: tests/python/unittest/test_numpy_op.py [hunk 1]
```diff
@@ -1650,6 +1650,7 @@ def hybrid_forward(self, F, a, b, *args, **kwargs):
         'power': (1.0, 2.0, [lambda y, x1, x2: _np.power(x1, x2 - 1.0) * x2],
                              [lambda y, x1, x2: _np.power(x1, x2) * _np.log(x1)]),
         'lcm': (-100, 100, [None], None, [[_np.int32]]),
+        'bitwise_xor': (-100, 100, [None], None, [[_np.int32]]),
```
Comment (reminisce) [inline tests/python/unittest/test_numpy_op.py]: Please add `bitwise_xor` to `_NUMPY_ARRAY_UFUNC_LIST` and test cases from the official numpy codebase to `test_numpy_interoperability.py`.

File: src/operator/tensor/elemwise_binary_op.h [hunk 2]
```diff
@@ -474,6 +474,31 @@ class ElemwiseBinaryOp : public OpBase {
...
                                        std::vector<int> *out_attrs);
 
+  template<typename xpu, typename OP>
+  static void ComputeInt(const nnvm::NodeAttrs &attrs,
+                         const OpContext &ctx,
+                         const std::vector<TBlob> &inputs,
+                         const std::vector<OpReqType> &req,
+                         const std::vector<TBlob> &outputs) {
+    using namespace mxnet_op;
+    if (req[0] != kNullOp) {
```
Comment (reminisce) [inline src/operator/tensor/elemwise_binary_op.h]: To save then indentation for the following lines, you can return here if `req[0]` is `kNullOp`.

Comment (haojin2) [reply src/operator/tensor/elemwise_binary_op.h]: Doing this now for other functions as well in this file.

**Comment by haojin2** [comment]
@ptrendx

**Pull request merged**